### PR TITLE
Initial implementation of extra REV Hub features

### PR DESCRIPTION
**Pull request by Windwoes**
Initial implementations of:

 - Total module current sensor
 - Per-motor current sensors
 - Servo bus current sensor
 - GPIO current sensor
 - I2C current sensor
 - Internal temperature sensor
 - 12v monitor sensor
 - 5v monitor sensor

### CODE REVIEW

File: Hardware/src/main/java/org/openftc/hardware/extraRevHubFeatures/OpenRevHub.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.openftc.hardware.extraRevHubFeatures;
```
Comment (ItsHarper) [inline Hardware/src/main/java/org/openftc/hardware/extraRevHubFeatures/OpenRevHub.java]: I'd like to see this package named `org.openftc.hardware.rev`.

**Comment by ItsHarper** [comment]
I don't have access to a Rev hub right now, so I'm not going to be able to do much actual testing when reviewing this. I'm just going to be seeing if everything looks good, and that it does everything exactly the same way as the official features. I'm trusting your testing on this, so please spend some time trying to break it. @WardBenjamin, if you have access to one, I'd appreciate it if you'd spend some time kicking the tires of this as well. Thanks.

**Comment by ItsHarper** [comment]
@FROGbots-4634 Didn't you say that the servo current sensor was returning weird values? Is that fixed now?

**Comment by Windwoes** [comment]
@NoahAndrews That's what Karter said a while ago. I haven't had time to test it much.

**Comment by ItsHarper** [comment]
For anyone who wants to help with testing this, see the documentation here: https://openftc.github.io/erhf/ExtraRevHubFeatures

**Comment by ItsHarper** [comment]
One thing this needs is Javadoc comments on all new classes and methods, as well as `// Modified for OpenFTC` comments wherever existing files are modified.

**Comment by LegoF4** [comment]
I can confirm that the servo power draw is still bugged; it was registering a higher draw on the servo then the entire hub, and a value well above stall current.

**Comment by ItsHarper** [comment]
https://openftc.github.io/erhf/ExtraRevHubFeatures

**Comment by ItsHarper** [comment]
image

**Pull request closed without merging**